connection.go

```diff
@@ -727,11 +684,58 @@ func (conn *Connection) closeConnection(neterr error, forever bool) (err error)
 	return
 }
 
+func (conn *Connection) getDialTimeout() time.Duration {
+	dialTimeout := conn.opts.Reconnect / 2
+	if dialTimeout == 0 {
+		dialTimeout = 500 * time.Millisecond
+	} else if dialTimeout > 5*time.Second {
+		dialTimeout = 5 * time.Second
+	}
+	return dialTimeout
+}
+
+func (conn *Connection) runReconnects() error {
+	dialTimeout := conn.getDialTimeout()
+	var reconnects uint
+	var err error
+
+	for reconnects <= conn.opts.MaxReconnects {
+		now := time.Now()
+
+		ctx, cancel := context.WithTimeout(context.Background(), dialTimeout)
+		err = conn.createConnection(ctx)
+		cancel()
+
+		if err == nil {
+			return nil
+		}
+		if clientErr, ok := err.(ClientError); ok &&
+			clientErr.Code == ErrConnectionClosed {
+			return err
+		}
```

Comment:
if err != nil {
} else {
```suggestion
		if err != nil {
			if clientErr, ok := err.(ClientError); ok &&
				clientErr.Code == ErrConnectionClosed {
				return err
			}
		} else {
			return nil
		}
```